mq/inmemory/test.cc

```diff
@@ -147,3 +146,41 @@ TEST(InMemoryMQ, WaitOnOverflowTest) {
   std::set<std::string> messages(begin(c.messages_), end(c.messages_));
   EXPECT_EQ(messages.size(), 100u);
 }
+
+TEST(InMemoryMQ, DefaultConsumer) {
+  struct Storage {
+    std::string messages;
+    size_t count = 0u;
```

Comment:
Nah, not because it's not thread-safe. Because without `volatile` or `atomic`, the compiler mya choose to optimize the while-loop below away, as redundant.